extract_endpoint/tests/test_endpoint.py

```diff
@@ -84,6 +84,36 @@ def mock_send_to_tl(data: typing.Dict[str, str]) -> int:
     monkeypatch.setattr(endpoint, 'send_to_tl', mock_send_to_tl)
 
 
+@pytest.fixture()
+def thread_runner() -> typing.Generator:
+    yield endpoint.ThreadRunner
+    endpoint.ThreadRunner.join()
+
+
+def test_threadrunner(thread_runner: endpoint.ThreadRunner) -> None:
+    stay_asleep = True
+
+    def sleeper() -> None:
+        while stay_asleep:
+            pass
+
+    thread_runner.start_new_thread(sleeper)
+    assert thread_runner.is_thread_running()
+    stay_asleep = False
+    thread_runner.join()
+    assert not thread_runner.is_thread_running()
+
+
+def check_file_in_azure(azure_service: blob.BlockBlobService,
+                        azure_emulator_coords: azure_utils.StorageCoordinates,
+                        filename: str,
+                        contents: str) -> None:
+
+    assert filename in [blob.name for blob in azure_service.list_blobs(azure_emulator_coords.container)]
```

Comment:
Probably doesn't matter in this context, but for future reference a `set` comprehension would be better since `in` checks are O(1)

Reply:
yeah, hash tables ftw. In this case the list only contains 3 objects.